appinfo/routes.php

```diff
@@ -36,5 +36,16 @@
 			'url' => '/settings/verifyNewSecret',
 			'verb' => 'POST'
 		],
+	],
+	'ocs' => [
+		[
+			'name' => 'totp_api#validate',
+			'url' => '/api/v1/validate/{id}/{secret}',
+			'verb' => 'GET',
+			'requirements' => [
+				'id' => '.+',
```

Comment:
I'll need an explanation for these requirements.
For an url such as `/api/v1/validate/userId/verySecret` I expect "id => userId" and "secret => verySecret". I expect that "/" is used as separator. The only way I think this could go wrong is if either the id or the secret contains a "/".
If the secret contains a "/" such as in `/api/v1/validate/userId/very/top/Secret` I expect "id => userId" and "secret => very/top/Secret", which might be fair (I don't know what chars are used as secret, but having "/" as part of the set doesn't seem a good idea). However, the system could also assume "id => userId/very/top" and "secret => Secret" for the same url.

Reply:
You are right about the secret keyword, I will update code to use `totp` keyword instead of `secret` and use `uid` instead of `id`.
However, Totp is 6 digit number, can not contain "/", also only the following characters are allowed in a username: `"a-z", "A-Z", "0-9", and "_.@-'` . So, it can not contain "/" too. IMHO, there is no risk in url.